presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -322,7 +354,7 @@ private static Page filter(Page page, Block mask)
     }
 
     private static class DistinctingGroupedAccumulator
```

Comment:
If we don't wanna use too much memory for this method, we can always call `compact()` to materialize the mask. This is true for `filter`. Haven't read the `AccumulatorCompiler` in details, but might also be fine for `distinctMask` as well. This trick has been used in many place (e.g, `GroupedTopNBuilder`)

Reply:
Did a few interface changes to accomplish this. Feel free to comment.